utils/installer/install.sh

```diff
@@ -31,13 +39,36 @@ declare -a __pip_deps=(
   "pynvim"
 )
 
+function parse_arguments() {
+  while (("$#")); do
+    case "$1" in
+      -y | --yes)
+        ARGS_INSTALL_NONINTERACTIVE="y"
+        ;;
+      -n | --no)
```

Comment:
`--no` is redundant and will break compatibility. It's enough by declaring `ARGS_INSTALL_NONINTERACTIVE=''` and checking if it's empty or not.

Reply:
I am not sure why you say `--no` is redundant. It is answering now for all interactive questions, which needs some mechanism to be signalled.

Also internally, the `ARGS_INSTALL_NONINTERACTIVE` has to have three values:
1.  Non-interactive and "yes to all". Here we have `ARGS_INSTALL_NONINTERACTIVE=y`
2. Non-interactive and  "no to all". Here we have `ARGS_INSTALL_NONINTERACTIVE=n`
3. Interactive. Here we have `ARGS_INSTALL_NONINTERACTIVE` not set
 
Maybe I do not understand how this can be achieved with two states.

Reply:
I guess I just don't see the use-case for it. I know that `-y` is pretty common, but it looks odd to me even looking at it from a pure logical point of view: `non-interactive=no` -> `not non-interactive` -> `interactive?`.
From a user-experience point of view: you pass `-y` if you trust the installer wouldn't do anything weird, but you wouldn't pass `-n` because you probably don't know what that means/does yet.
Maybe you didn't notice that the answers to these questions are set to `no` by default? you can just quickly press `enter` and move on with the installation.

Since this is user-experience oriented, let's see what the others think
cc. @abzcoding, @Tastyep, @chaesngmin

Reply:
IMO, the interactive option isn't necessary. I see the usage for `apt` where you can do `apt install --yes`, but for the setup script since I'm only executing it once or twice I don't really see a use case.

Reply:
I think it's useful because it allows running the setup script in non-interactive mode as part of a script.

I think this is useful to package the editor setup as a container. I do this to make my editor/dotfiles portable across machines. Having this in the installer script allows the install to be consistent, as opposed doing a manual install in the dockerfile. 

It seems that in the current container setup (i.e., `utils/docker`) will fail because of the interactive input requirement of `install.sh`. (**In fact, it currently fails because it can't detect Neovim. But once that is fixed, it will fail because of the input requirement.**)

In any case, I can get rid of the `--no` if you guys think that it isn't useful.